### DISCUSSION TITLE
extract data value along a line

### DISCUSSION THREAD

**Discussion by tnn77** [raphaelquast/EOmaps#54] [Q&A] [answered]
@raphaelquast 
I stumbled across EOmaps when I was looking for an easy way to add a scalebar to stereographic projection figures that I often create ("Adding scalebars - what about distances?").

I am really interested in using your software as I spend a lot of time playing around with metocean data map figures for polar research activities (i.e., making nice plots is essential, but that is not the essence of my interest). 

I tried your examples, and they worked out of the box for me, which is makes me more interested in using EOmaps. Now, I browsed the Documentation but was not sure if there is a feature to extract a gridded data along a line (e.g., data values along a line in your gridded data shown in "Turn your maps into powerful widgets"). If so, would it be possible to add to an example in the EOmaps examples?

Thank you in advance for your consideration.

**Comment by raphaelquast** [comment, marked as answer]
@tnn77 
Hey, thanks for checking out EOmaps and for the interesting suggestion!

Since your problem is in fact a nice general use-case, I've added a new example that shows how you can extract whole rows or columns of data when clicking on the map (make sure to update to EOmaps v3.1.4):

 - [🧮 Select 1D slices of 2D datasets](https://eomaps.readthedocs.io/en/latest/EOmaps_examples.html#select-1d-slices-of-a-2d-dataset) 


... if you create some nice widgets that might be useful for others as well, feel free to open a pull-request to add it to the docs!

**Reply by tnn77** [reply]
@raphaelquast 
wow, thank you so much. I'd like to draw arbitrary lines, but the example is enough for me to work off. 

I am going to try find/make time to learn EOmaps  and hopefully I can use them as part of my (routine) work. 

Thank you so much, and hope to be in touch again.

**Reply by raphaelquast** [reply]
hey, 
no problem... glad I could help!

Feel free to suggest further ideas for usability-upgrades or features! 
I try my best to make EOmaps as accessible as possible to simplify data-analysis workflows for various use-cases.
